Program/Tinkering-Audio

```diff
@@ -0,0 +1,186 @@
+def SliceNSwap():
+  seconds = 18
+  targetIndex = 1
+  targetSound = makeEmptySound(int(sampleRate * seconds), sampleRate)
+
+  
+  #Get the second half of the sound and paste it into the end of the empty target sound
+  for i in range(getLength(LongSound)/2, 749690):
+    value = getSampleValueAt(LongSound, i)
+    setSampleValueAt (targetSound, targetIndex, value)
+    targetIndex = targetIndex +1
+    
+   
+  #Get the disconnect sound and paste it into the middle of the empty target sound
+  for i in range(0, getLength(DisconnectSound)):
+    value = getSampleValueAt(DisconnectSound, i)
+    setSampleValueAt (targetSound, targetIndex, value)
+    targetIndex = targetIndex +1
+
+  #Get the first half of the sound and paste it into the start of the empty target sound
+  for i in range(0, getLength(LongSound)/2):
+    value = getSampleValueAt(LongSound, i)
+    setSampleValueAt (targetSound, targetIndex, value)
+    targetIndex = targetIndex +1
+
+  explore(targetSound)
+  play(targetSound)
+  
+
+
+
+
+#Audio envilopes and echos
+
+def Echo(delay, sound):
+  soundsamples = getSamples(sound)
+  for index in range(len(soundsamples)-delay,0,-1):
+    value1 = getSampleValue(soundsamples[index])
+    value2 = getSampleValue(soundsamples[index-delay])
+    setSampleValue(soundsamples[index], value1+value2)
+  normalize(LongSound)
+  play(sound)
+  explore(sound)
+
+
+#ADS envelope
+def Envelope(tone, length, frequency, volume): #Envelope(tone1(), 3, frequency, volume)
+  seconds = 18
+  targetIndex = 1
+  attack_time = int(length * 0.2 * sampleRate)
+  sustain_time = int(length * 0.5 * sampleRate)
+  release_time = int(length * 0.3 * sampleRate)  
+ 
+    
+  interval = 1.0 / frequency
+  samplesPerCycle = interval * sampleRate
+  maxCycle = 2 * pi
+  tone = makeEmptySoundBySeconds(length)
+  
+#Get the first 10th of the tone and increase amplitude
+  for pos in range(0, getLength(tone)/10):
+    rawSample = sin((pos / samplesPerCycle) * maxCycle)
+    sampleVal = int(volume * rawSample)
+    volume = volume +1
+    setSampleValueAt(tone, pos, sampleVal)    
+
+#Between the first 10th and 8th of the tone decrease amplitude
+  for pos in range(getLength(tone)/10, getLength(tone)/8):
+    rawSample = sin((pos / samplesPerCycle) * maxCycle)
+    sampleVal = int(volume * rawSample)
+    volume = volume -1
+    setSampleValueAt(tone, pos, sampleVal)
+
+#From an 8th till the end play the same amplitude
+  for pos in range(getLength(tone)/8, getLength(tone)):
+    rawSample = sin((pos / samplesPerCycle) * maxCycle)
+    sampleVal = int(volume * rawSample)
+    setSampleValueAt(tone, pos, sampleVal)
+  
+
+  explore(tone)
+  play(tone)
+  
+
+  
+#Parsing tokens into audio
+def parser(bitDepth):
+  noteiteration = 0
+  notes = requestString("Please enter some notes such as; C, D, E, F, G, B, A, C") 
+  
+  for x in notes:
+    notes.split(' ')
+    frequency = search(noteNum, '%s' % notes[noteiteration])
+    noteiteration = noteiteration +1
```

Comment:
This section of code requires clarification. It is not clear how frequencies are selected.